src/main/java/org/jabref/gui/ClipBoardManager.java

```diff
@@ -29,35 +34,43 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-public class ClipBoardManager {
-    public static final DataFormat XML = new DataFormat("application/xml");
+// Singleton with enum
+public enum ClipBoardManager {
+
+    INSTANCE;
 
     private static final Logger LOGGER = LoggerFactory.getLogger(ClipBoardManager.class);
 
-    private final Clipboard clipboard;
-    private final ImportFormatReader importFormatReader;
+    private final Clipboard clipboard = Clipboard.getSystemClipboard();
+    private final java.awt.datatransfer.Clipboard primary = Toolkit.getDefaultToolkit().getSystemSelection();
+    private final ImportFormatReader importFormatReader = Globals.IMPORT_FORMAT_READER;
 
-    public ClipBoardManager() {
-        this(Clipboard.getSystemClipboard(), Globals.IMPORT_FORMAT_READER);
+    ClipBoardManager() {
     }
 
-    public ClipBoardManager(Clipboard clipboard, ImportFormatReader importFormatReader) {
-        this.clipboard = clipboard;
-        this.importFormatReader = importFormatReader;
+    public void watchInput(String newValue) {
+        if (!newValue.isEmpty()) {
+            primary.setContents(new StringSelection(newValue), null);
+        }
     }
 
-    /**
-     * Puts content onto the clipboard.
-     */
-    public void setContent(ClipboardContent content) {
-        clipboard.setContent(content);
+    public void inputToPrimary(TextInputControl input) {
+        input.insertText(input.getCaretPosition(), getContentsPrimary());
     }
 
+    // For activating this functionality in an input, use:
+    //
+    // input.selectedTextProperty().addListener((observable, oldValue, newValue) -> watchInput(newValue));
```

Comment:
It would be nice if these setup steps could be simplified and combined into one method, so that it is enough to call `ClipBoardManager.addX11Support(input)`. This also has the advantage that you can inline `watchInput` and `inputToPrimary`.

Reply:
Yep. I did something like that in the first version...

https://github.com/JabRef/jabref/blob/a54ddd174ac8c0f3906625cc77824fe1f382f68e/src/main/java/org/jabref/gui/ClipBoardManager.java#L73-L84

But then I spent many hours trying to solve some errors that appeared when doing `./gradlew test`. As I wanted to have an available _bugs free_ version, I made those changes; but I will keep improving and testing this stuff in the next few days.

**:"D**

Reply:
That could also be a problem with our gui tests. There are a bit unreliably (to phrase it politely).